### PR TITLE
- replace runtime for RelWithDebInfo & MinSizeRel configurations

### PR DESCRIPTION
**Pull request by SSE4**
follow up #1854

### CODE REVIEW

File: conans/client/generators/cmake_common.py [hunk 1]
```diff
@@ -458,18 +458,18 @@ def generate_targets_section(dependencies):
...
-            string(REPLACE "/MDd" ${CONAN_LINK_RUNTIME} CMAKE_C_FLAGS_DEBUG ${CMAKE_C_FLAGS_DEBUG})
-        endif()
+        foreach(flag CMAKE_C_FLAGS_RELEASE CMAKE_CXX_FLAGS_RELEASE
+                     CMAKE_C_FLAGS_RELWITHDEBINFO CMAKE_CXX_FLAGS_RELWITHDEBINFO
+                     CMAKE_C_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_MINSIZEREL)
+            if(DEFINED ${flag})
+                string(REPLACE "/MD" ${CONAN_LINK_RUNTIME} ${flag} "${${flag}}")
+            endif()
+        endforeach()
+        foreach(flag CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_DEBUG)
```
Comment (theodelrieu) [inline conans/client/generators/cmake_common.py]: Shouldn't it be `CMAKE_C_FLAGS_DEBUG`?

File: conans/client/generators/cmake_common.py [hunk 2]
```diff
@@ -524,18 +524,18 @@ def generate_targets_section(dependencies):
...
-            string(REPLACE "/MDd" "/MTd" CMAKE_C_FLAGS_DEBUG ${CMAKE_C_FLAGS_DEBUG})
-        endif()
+        foreach(flag CMAKE_C_FLAGS_RELEASE CMAKE_CXX_FLAGS_RELEASE
+                     CMAKE_C_FLAGS_RELWITHDEBINFO CMAKE_CXX_FLAGS_RELWITHDEBINFO
+                     CMAKE_C_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_MINSIZEREL)
+            if(DEFINED ${flag})
+                string(REPLACE "/MD" "/MT" ${flag} "${${flag}}")
+            endif()
+        endforeach()
+        foreach(flag CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_DEBUG)
```
Comment (theodelrieu) [inline conans/client/generators/cmake_common.py]: same here

File: conans/client/generators/cmake_common.py (hunk 1 above)
Comment (SSE4) [reply conans/client/generators/cmake_common.py]: good catch, will fix

Comment (theodelrieu) [reply conans/client/generators/cmake_common.py]: I opened #1863 in the meantime ;)

**Pull request merged**